scripts/prettier.js

```diff
@@ -6,62 +6,98 @@
+  } else {
+    prefix = '**/';
+  }
 
-if (!files.length) {
-  process.exit(0);
+  if (!path.extname(file)) {
+    suffix = '/**';
+  }
+  console.log(`${prefix}${hasSlashPrefix ? file.substring(1) : file}${suffix}`);
+  return `${prefix}${hasSlashPrefix ? file.substring(1) : file}${suffix}`;
 }
 
-files.forEach(file => {
-  const options = prettier.resolveConfig.sync(file, {
-    config: prettierConfigPath,
-  });
-  try {
-    const input = fs.readFileSync(file, 'utf8');
-    if (shouldWrite) {
-      console.log(`Formatting ${file}`);
-      const output = prettier.format(input, { ...options, filepath: file });
-      if (output !== input) {
-        fs.writeFileSync(file, output, 'utf8');
-      }
-    } else if (!prettier.check(input, { ...options, filepath: file })) {
-      if (!didWarn) {
-        console.log(`\nThis project uses prettier to format all JavaScript code.
-           Please run 'yarn prettier-all' and add changes to files listed
-           below to your commit:\n\n`);
-        didWarn = true;
+function runPrettier(changedFiles) {
+  let didWarn = false;
+  let didError = false;
+
+  const warnedFiles = [];
+  const ignoredFiles = fs
+    .readFileSync('.eslintignore', 'utf-8')
+    .split(/\r*\n/)
+    .filter(notEmpty => notEmpty)
+    .map(file => mapEslintEntriesToGlobs(file));
+
+  const files = glob
+    .sync('**/*.{js,tsx,d.ts}', { ignore: ['**/node_modules/**', ...ignoredFiles] })
+    .filter(f => !changedFiles || changedFiles.has(f));
+
+  if (!files.length) {
+    process.exit(0);
+  }
+
+  files.forEach(file => {
+    const options = prettier.resolveConfig.sync(file, {
+      config: prettierConfigPath,
+    });
+    try {
+      const input = fs.readFileSync(file, 'utf8');
+      if (shouldWrite) {
+        console.log(`Formatting ${file}`);
+        const output = prettier.format(input, { ...options, filepath: file });
+        if (output !== input) {
+          fs.writeFileSync(file, output, 'utf8');
+        }
+      } else {
+        console.log(`Checking ${file}`);
+        if (!prettier.check(input, { ...options, filepath: file })) {
+          warnedFiles.push(file);
+          didWarn = true;
+        }
       }
+    } catch (error) {
+      didError = true;
+      console.log(`\n\n${error.message}`);
       console.log(file);
     }
-  } catch (error) {
-    didError = true;
-    console.log(`\n\n${error.message}`);
-    console.log(file);
+  });
+
+  if (didWarn) {
+    console.log(
+      '\n\nThis project uses prettier to format all JavaScript code.\n' +
+        `Please run '${!changedFiles ? 'yarn prettier:all' : 'yarn prettier'}'` +
+        'and commit the changes to the files listed below:\n\n',
+    );
+    console.log(warnedFiles.join('\n'));
   }
-});
 
-if (didWarn || didError) {
-  process.exit(1);
+  if (didWarn || didError) {
+    process.exit(1);
+  }
+}
+
+if (onlyChanged) {
+  listChangedFiles()
+    .then(changedFiles => {
+      runPrettier(changedFiles);
```

Comment:
I'd rather be safe than sorry here and return the result. Even if `runPrettier` is sync now if we make it async this will cause a common promise pitfall: https://github.com/xjamundx/eslint-plugin-promise/blob/master/docs/rules/always-return.md